http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -134,7 +135,16 @@ object KubernetesServiceDbQueries {
     joinFullAppAndUnmarshal(kubernetesClusterQuery.findByIdQuery(id), nodepoolQuery, appQuery)
       .map(_.headOption)
 
-  def markPendingCreating(nodepoolId: NodepoolLeoId, appId: AppId, createCluster: Option[CreateCluster])(
+  def getAllNodepoolsForCluster(
+    clusterId: KubernetesClusterLeoId
+  )(implicit ec: ExecutionContext): DBIO[List[Nodepool]] =
+    for {
+      nodepools <- nodepoolQuery
+        .findActiveByClusterIdQuery(clusterId)
+        .result
+    } yield nodepools.map(rec => unmarshalNodepool(rec, List.empty)).toList
```

Comment:
if we need to sub out `List.empty`, can we just return the actual data type we're retrieving?

Reply:
its more expensive to include the list since its apps, and apps require a lot f joining (app, disk, label, service, namespace). Only af few queries need all the joins

Reply:
I understand joining is costly....I meant to return the data type without `List`...like straight-up db record instead